### PR TITLE
Add support for source line tracking through synthesis phase

### PR DESCRIPTION
**Pull request by jlowder**
These are some minor edits to help retain source line tracking information on cells through some of the synthesis passes, i.e. propagate "src" attributes from original cells to the derived cells they are being replaced with.

### CODE REVIEW

File: kernel/rtlil.cc [hunk 1]
```diff
@@ -1620,18 +1620,19 @@ RTLIL::Cell *RTLIL::Module::addCell(RTLIL::IdString name, const RTLIL::Cell *oth
 }
 
 #define DEF_METHOD(_func, _y_size, _type) \
-	RTLIL::Cell* RTLIL::Module::add ## _func(RTLIL::IdString name, RTLIL::SigSpec sig_a, RTLIL::SigSpec sig_y, bool is_signed) { \
+		RTLIL::Cell* RTLIL::Module::add ## _func(RTLIL::IdString name, RTLIL::SigSpec sig_a, RTLIL::SigSpec sig_y, bool is_signed, std::string src) { \
```
Comment (cliffordwolf) [inline kernel/rtlil.cc]: Why the indent change? Imo it was correct before.

File: passes/techmap/alumacc.cc [hunk 2]
```diff
@@ -55,27 +55,27 @@ struct AlumaccWorker
 
 		RTLIL::SigSpec get_gt() {
 			if (GetSize(cached_gt) == 0)
-				cached_gt = alu_cell->module->Not(NEW_ID, alu_cell->module->Or(NEW_ID, get_lt(), get_eq()));
+				cached_gt = alu_cell->module->Not(NEW_ID, alu_cell->module->Or(NEW_ID, get_lt(), get_eq()), false, alu_cell->attributes["\\src"].decode_string());
```
Comment (cliffordwolf) [inline passes/techmap/alumacc.cc]: Commit 8530333 adds `get_src_attribute()` and `set_src_attribute()`. Using `get_src_attribute()` has the advantage that it does not create an empty src attribute on objects that do not have one yet. Please update your PR to use this two functions.

Comment (jlowder) [reply passes/techmap/alumacc.cc]: Sounds good. I have updated the PR to use the new functions - please let me know if anything else is needed.

**Comment by cliffordwolf** [comment]
I've merged the PR and changed the remaining places where the new functions should be used in the new code in commit 8a66bd3.

**Pull request merged**